### PR TITLE
feat: code parity with other native providers

### LINKED ISSUE #9: Update structure to mirror azure/kubernetes repos
In both native providers, we have a `provider` directory which contains `pkg` etc. We should make sure these boilerplate repo mirrors that experience.

### PR DESCRIPTION
**Pull request by frezbo**
Fixes #9 and #3

Signed-off-by: Noel Georgi <git@frezbo.dev>

### CODE REVIEW

File: Makefile
Comment (mikhailshilkov) [inline Makefile:7]:
```suggestion
NUGET_PKG_NAME   := Pulumi.Xyz
```

File: Makefile
Comment (mikhailshilkov) [inline Makefile:12]: A brand-new provider shouldn't start with v3
```suggestion
PROVIDER_PATH   := provider
```

File: Makefile
Comment (mikhailshilkov) [inline Makefile:90]: I believe we don't have any integration tests, so these commands make little sense

**Review by mikhailshilkov — COMMENTED** [review]
This looks great, thank you for taking the time to refactor this! A few small comments below.

File: Makefile
Comment (frezbo) [reply Makefile]: fixed

File: Makefile
Comment (frezbo) [reply Makefile]: fixed

File: Makefile
Comment (frezbo) [reply Makefile]: fixed

**Comment by frezbo** [comment]
@mikhailshilkov fixed the review comments. Thanks for taking a look :hugs:

**Review by mikhailshilkov — APPROVED** [review]
LGTM. @viveklak do you want to take I look before I merge?

**Comment by lukehoban** [comment]
BTW - @stack72 - there are a number of places where the Makefile here differs from what we have in all our other boilerplates.  It's probably a future follow-up thing, not something we need to block this PR on, but it would be good to rationalize across the boilerplate repos (for both native providers and components) since both should in principle be effectively "the same" from a Makefile perspective.  We should likely then push that concsisten approach back up into some of the 1st party native providers so we have a consistent approach to how providers are built/published.

Comparison: https://github.com/pulumi/pulumi-component-provider-go-boilerplate/blob/main/Makefile

**Comment by lukehoban** [comment]
I think this is good to merge now as a first step.  @viveklak or @stack72 anything you wanted to review before landing this?  I do expect we may want some follow-up to standardize a little more between the component and native provider boilerplates, but that can easily be follow up.

**Comment by frezbo** [comment]
@lukehoban oops I missed the comments. I was mostly basing the makefile from the native providers. I forgot to look at the ones in the component providers. That could be done as a separate PR

**Pull request merged**
